src/lib/xof/xof.h

```diff
@@ -158,6 +158,16 @@ class BOTAN_PUBLIC_API(3, 2) XOF {
          return out;
       }
 
+      /**
+       * @return the next @p count output bytes as a std::array<>.
+       */
+      template <size_t count>
+      std::array<uint8_t, count> output() {
+         std::array<uint8_t, count> out;
+         generate_bytes(out);
+         return out;
+      }
+
```

Comment:
This type of "data source" method is really useful, we should probably add it to other similar classes. E.g. the RNG. That would allow things like: `uint32_t i = load_le(rng.random_array<4>())`. Now this would be:

```C++
std::array<uint8_t, 4> buffer;
rng.randomize(buffer);
uint32_t i = load_le(buffer);
```

Reply:
See #4152